storage/api.go

```diff
@@ -0,0 +1,95 @@
+package storage
+
+import (
+	"context"
+	"io"
+)
+
+// --- basics --->
+
+type ReadableStorage interface {
+	Get(ctx context.Context, key string) ([]byte, error)
```

Comment:
What's the precedent for using `key string` rather than `key []byte`? Basically every key-value database in Go (badger, bolt, pebble, etc etc) use `[]byte` keys, precisely because it helps clarify that they can contain anything - not just valid utf8.

If the point is to be more like go-datastore, perhaps that's enough precedent. But I would hope that we would make the interface be what is nicest for Go and as a generic abstraction layer, not just for IPFS in particular.